service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java

```diff
@@ -133,14 +137,21 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(
 
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
-        @RequestBody CallbackRequest callbackRequest) {
+        @RequestBody CallbackRequest callbackRequest,
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId) throws IOException {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
+        Document document = getDocument(authorization, userId, caseData, SEALED);
```

Comment:
there was some logic we have in sdo and cmo for drafts, is any of this reusable?

Reply:
both them methods vary a tiny bit in themselves, this one is the same in that we need the specific order title grabbed from order doc and subtype.. i can changed 'SEALED' to be a boolean of draft/not draft if you would prefer but I think this is fine as is